src/pybind/mgr/volumes/fs/async_cloner.py

```diff
@@ -265,24 +242,29 @@ def handle_clone_in_progress(fs_client, volspec, volname, index, groupname, subv
 
 def handle_clone_failed(fs_client, volspec, volname, index, groupname, subvolname, should_cancel):
     try:
-        with open_volume(fs_client, volname) as fs_handle:
-            # detach source but leave the clone section intact for later inspection
-            with open_clone_subvolume_pair(fs_client, fs_handle, volspec,
-                                           volname, groupname, subvolname) \
-                as (subvol0, subvol1, subvol2):
-                subvol1.detach_snapshot(subvol2, index)
+        # detach source but leave the clone section intact for later inspection
+        with open_clone_sv_pair_in_vol(fs_client, volspec, volname, groupname,
+                subvolname) as (subvol0, subvol1, subvol2):
+            subvol1.detach_snapshot(subvol2, index)
     except (MetadataMgrException, VolumeException) as e:
         log.error("failed to detach clone from snapshot: {0}".format(e))
     return (None, True)
 
 def handle_clone_complete(fs_client, volspec, volname, index, groupname, subvolname, should_cancel):
     try:
-        with open_volume(fs_client, volname) as fs_handle:
-            with open_clone_subvolume_pair(fs_client, fs_handle, volspec,
-                                           volname, groupname, subvolname) \
-                as (subvol0, subvol1, subvol2):
-                subvol1.detach_snapshot(subvol2, index)
-                subvol0.remove_clone_source(flush=True)
+        # XXX: Not using lock will result in race condition between cloner
+        # threads to write to subvolume's metadata file (which is named as
+        # ".meta"). Thus, updates written to subvolume metadata by previous
+        # cloner thread will get deleted due over-writing.
```

Comment:
nit: s/due over-writing/due to over-writing

Reply:
Done.